arch/risc-v/src/common/riscv_exception_common.S

```diff
@@ -91,24 +93,12 @@ exception_common:
   mv         a1, sp               /* context = sp */
 
 #if CONFIG_ARCH_INTERRUPTSTACK > 15
-
-  /* Offset to hartid */
-
-  mv         s0, a0               /* save cause */
-  jal        x1, riscv_mhartid    /* get hartid */
-
   /* Switch to interrupt stack */
 
-#if IRQ_NSTACKS > 1
-  li         t0, (CONFIG_ARCH_INTERRUPTSTACK & ~15)
-  mul        t0, a0, t0
-  la         a0, g_intstacktop
-  sub        sp, a0, t0
-#else
-  la         sp, g_intstacktop
-#endif
-
-  mv         a0, s0               /* restore cause */
+  mv         t0, a0               /* save cause */
+  jal        x1, riscv_percpu_get_irqstack
```

Comment:
This could be wrapped so that the percpu module is not a strict requirement here